pkg/cli/cli_test.go

```diff
@@ -0,0 +1,344 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/cmd"
+	"k8s.io/kubectl/pkg/cmd/plugin"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+func TestOCSubcommandShadowPlugin(t *testing.T) {
```

Comment:
I assume we will not merge this PR until the upstream PR is merged in order to sync them to the latest version?

Reply:
I think, I missed an important point. We are making changes on upstream 1.28. But our oc rebase is basically picking 1.27.1. That means we'll have to wait next rebase to move these changes. 
In our oc rebase PR, we should live with this patch for this release.

Reply:
I suppose, if we decide to sync and base it on upstream. If not, we could introduce it sooner.